File: backend/models/table.py
"""Pydantic schemas for table and column metadata."""
from typing import Optional
from pydantic import BaseModel


class ColumnMetadata(BaseModel):
    name: str
    data_type: str
    is_nullable: bool = True
    is_primary_key: bool = False
    is_foreign_key: bool = False
    foreign_key_ref: Optional[str] = None   # "other_table.column"
    ai_description: Optional[str] = None


class TableMetadata(BaseModel):
    table_name: str
    fully_qualified_name: str
    columns: list[ColumnMetadata]
    row_count: Optional[int] = None
    business_summary: Optional[str] = None
    usage_recommendations: Optional[str] = None
    is_documented: bool = False

File: frontend/src/types/table.ts
/** Table and column metadata types */
export interface ColumnMetadata {
    name: string;
    data_type: string;
    is_nullable: boolean;
    is_primary_key: boolean;
    is_foreign_key: boolean;
    foreign_key_ref?: string;
    ai_description?: string;
}

export interface TableMetadata {
    table_name: string;
    fully_qualified_name: string;
    columns: ColumnMetadata[];
    row_count?: number;
    business_summary?: string;
    usage_recommendations?: string;
    is_documented: boolean;
}
